.github/workflows/publish-exp.yml

```diff
@@ -11,13 +11,14 @@ jobs:
   build:
     runs-on: ubuntu-24.04
     steps:
+      - uses: actions/checkout@v4
+      - uses: gradle/actions/wrapper-validation@v4
       - name: setup jdk
         uses: actions/setup-java@v4
         with:
           java-version: 21
           distribution: 'temurin'
-      - uses: actions/checkout@v4
-      - uses: gradle/actions/wrapper-validation@v4
+      - uses: gradle/actions/setup-gradle@v4
```

Comment:
Never use caching for publish runs that contain the secrets, its not clear where it ends up.